scan/tls_handshake.go

```diff
@@ -105,21 +188,35 @@ func cipherSuiteScan(host string) (grade Grade, output Output, err error) {
 		ciphers := make([]uint16, len(allCiphers))
 		copy(ciphers, allCiphers)
 		for len(ciphers) > 0 {
-			cipherIndex, err := sayHello(host, ciphers, vers)
+			var cipherIndex int
+			cipherIndex, _, err = sayHello(host, ciphers, nil, vers)
 			if err != nil {
-				break
+				if err == tls.HandshakeFailureErr {
+					err = nil
+					break
+				}
+				return
```

Comment:
In non `alertHandshakeFailure` situations, we still treat the `err` from `sayHello` as a "true error". _Any_ TLS alert should just be ignored and treated with a break from the inner loop, not a return.

You're right that we probably shouldn't log these alerts, since they're expected. It's probably simplest to add a sentinel value inside the `scan` package.